src/renderers/WebGLRenderer.js

```diff
@@ -1720,6 +1723,31 @@ function WebGLRenderer( parameters = {} ) {
 		p_uniforms.setValue( _gl, 'normalMatrix', object.normalMatrix );
 		p_uniforms.setValue( _gl, 'modelMatrix', object.matrixWorld );
 
+		// UBOs
+
+		if ( material.isShaderMaterial || material.isRawShaderMaterial ) {
+
+			const groups = material.uniformsGroups;
```

Comment:
@Mugen87 I had an exception on this line using a package which had an old [TypeScript types of three.js (134)
](https://www.npmjs.com/package/@types/three/v/0.134.0) I'm not sure how it works but seems like something along the way stripped this member from the RawSahderMaterial instance (internally in package this member was initialized correctly to an `[]`)
`uniformsGroups` didn't exist on the instance at all, so the `groups.length` was the actual exception
So I feel like there's a backwards compatibility issue here but I can't create a JSFiddle with an example

Reply:
That is strange since UBO support was added with `r143` and all related classes and properties were added with this release. 

Sorry, but we can't backwards compatibility for use cases when external third party libs prematurely add support for work-in-progress code.

Reply:
I'll clarify, I use threejs 143 (no typescript in the project)
I import a package that's using 134+ts
The package somehow creates ShaderMaterials without this property
So the package is not ahead of my project - which I think should be supported
I'll try to work tomorrow on a jsfiddle, but I'm really not sure how I'll import 2 versions of threejs :)

Reply:
> but I'm really not sure how I'll import 2 versions of threejs :)

That would not be a valid workflow anyway.

Reply:
I patched the package on our private copy, so I don't have the issue anymore (doing what you mentioned)
But I believe that given 2 packages **A** and **B** where:
- **A** uses typescript and is using three.js r134 (or anything below 143), and if creating objects to be rendered by **B**.
- **B** uses three.js r143 or up (and is not using typescript? not sure it matters), **B** is running the main rendering loop and is responsible to create the canvas and renderer etc.
**A** will produce objects which are illegal in **B** (since this field doesn't exist in r134).
I have very little knowledge on how A and B are interacting with one another, might be a misconfiguration on that particular package and how it's imported.

Reply:
Seems like I'm wrong, this example shows the same scenario:
https://codesandbox.io/s/pix4d-three-potree-loader-example-forked-s6i6i1?file=/package.json

[The imported package uses r139](https://github.com/pnext/three-loader/blob/master/package.json#L30)
But the example uses r143 without a problem

Reply:
It might work to combine `three.js` from different releases but officially it is not supported. If library A creates an instance of `THREE.Mesh` from `r100` there is absolutely no guarantee that `r143` can process it. Let's be honest, it's quite unrealistic to support such a scenario.